diff --git a/cmake/BuildFlags.cmake b/cmake/BuildFlags.cmake
index d344e4ae..0ff30ba7 100644
--- a/cmake/BuildFlags.cmake
+++ b/cmake/BuildFlags.cmake
@@ -31,9 +31,9 @@ function(CHECK_SYCL_FLAG FLAG VARIABLE_NAME)
 endfunction()
 
 macro(set_build_flags)
- set(_TORCH_XPU_OPS_SUPPORTED_COMPILERS "GNU" "MSVC")
+ set(_TORCH_XPU_OPS_SUPPORTED_COMPILERS "Clang" "GNU" "MSVC")
   if(NOT CMAKE_CXX_COMPILER_ID IN_LIST _TORCH_XPU_OPS_SUPPORTED_COMPILERS)
-    message(WARNING "Not compiling with XPU. Currently only support GCC compiler on Linux and MSVC compiler on Windows as CXX compiler.")
+    message(WARNING "Not compiling with XPU. Currently only support Clang or GCC on Linux and MSVC on Windows as CXX compiler.")
     return()
   endif()
   set(SYCL_HOST_FLAGS)
@@ -70,7 +70,7 @@ macro(set_build_flags)
     set(SYCL_HOST_PER_CONFIG_FLAGS
       $<$<CONFIG:Debug>:/Od;/Z7>
       $<$<CONFIG:RelWithDebInfo>:/O2;/Z7>)
-  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
     list(APPEND SYCL_HOST_FLAGS -fPIC)
     list(APPEND SYCL_HOST_FLAGS -std=${CPP_STD})
     # Excluding warnings which flood the compilation output
@@ -138,7 +138,7 @@ macro(set_build_flags)
     list(APPEND SYCL_KERNEL_OPTIONS /Qftz-)
     # Suppress warnings about dllexport.
     list(APPEND SYCL_KERNEL_OPTIONS -Wno-ignored-attributes)
-  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
     list(APPEND SYCL_KERNEL_OPTIONS -std=${CPP_STD})
     list(APPEND SYCL_KERNEL_OPTIONS -Wno-absolute-value)
     # -fma which we used before is an alias used for -ffp-contract=fast for compatibility reasons
diff --git a/test/sycl/CMakeLists.txt b/test/sycl/CMakeLists.txt
--- a/test/sycl/CMakeLists.txt
+++ b/test/sycl/CMakeLists.txt
@@ -33,7 +33,7 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
   # Windows
   target_link_libraries(test_sycl_build_archive
   "-WHOLEARCHIVE:\"$<TARGET_FILE:sycl_simple_kernel_test>\"")
-elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
   # Linux
   target_link_libraries(test_sycl_build_archive
     "-Wl,--whole-archive,\"$<TARGET_FILE:sycl_simple_kernel_test>\" -Wl,--no-whole-archive")
