diff --git a/src/ATen/native/xpu/sycl/UnaryLogKernels.cpp b/src/ATen/native/xpu/sycl/UnaryLogKernels.cpp
index 0ac15509..69b531d6 100644
--- a/src/ATen/native/xpu/sycl/UnaryLogKernels.cpp
+++ b/src/ATen/native/xpu/sycl/UnaryLogKernels.cpp
@@ -30,7 +30,7 @@ template <typename scalar_t>
 struct LogFunctor {
   scalar_t operator()(scalar_t a) const {
     if constexpr (c10::is_complex<scalar_t>::value) {
-      return std::log(a);
+      return c10::math::log(a);
     } else {
       using opmath_t = at::opmath_type<scalar_t>;
       return sycl::log(static_cast<opmath_t>(a));
@@ -66,7 +66,7 @@ template <typename scalar_t>
 struct Log10Functor {
   scalar_t operator()(scalar_t x) const {
     if constexpr (c10::is_complex<scalar_t>::value) {
-      return std::log10(x);
+      return c10::math::log10(x);
     } else {
       using opmath_t = at::opmath_type<scalar_t>;
       return sycl::log10(static_cast<opmath_t>(x));
@@ -87,7 +87,7 @@ template <typename scalar_t>
 struct Log1pFunctor {
   scalar_t operator()(scalar_t x) const {
     if constexpr (c10::is_complex<scalar_t>::value) {
-      return std::log1p(x);
+      return c10::math::log1p(x);
     } else {
       using opmath_t = at::opmath_type<scalar_t>;
       return sycl::log1p(static_cast<opmath_t>(x));
@@ -108,7 +108,7 @@ template <typename scalar_t>
 struct Log2Functor {
   scalar_t operator()(scalar_t x) const {
     if constexpr (c10::is_complex<scalar_t>::value) {
-      return std::log2(x);
+      return c10::math::log2(x);
     } else {
       using opmath_t = at::opmath_type<scalar_t>;
       return sycl::log2(static_cast<opmath_t>(x));
diff --git a/src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp b/src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp
index 925959c9..c2635044 100644
--- a/src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp
+++ b/src/ATen/native/xpu/sycl/UnarySpecialOpsKernels.cpp
@@ -33,7 +33,7 @@ struct SigmoidFunctor {
     using opmath_t = at::opmath_type<scalar_t>;
     const auto one = opmath_t{1.0};
     if constexpr (c10::is_complex<opmath_t>::value) {
-      return one / (one + std::exp(-static_cast<opmath_t>(a)));
+      return one / (one + c10::math::exp(-static_cast<opmath_t>(a)));
     } else {
       return one / (one + sycl::exp(-static_cast<opmath_t>(a)));
     }
@@ -129,7 +129,7 @@ struct Exp2Functor<c10::complex<T>> {
   c10::complex<T> operator()(c10::complex<T> x) const {
     // There is no std::exp2 overload for complex, so instead
     // use the identity 2^x = e^(ln(2) * x)
-    return std::exp(std::numbers::ln2_v<T> * x);
+    return c10::math::exp(std::numbers::ln2_v<T> * x);
   }
 };
 
@@ -330,7 +330,7 @@ struct SincFunctor {
       using opmath_t = at::opmath_type<scalar_t>;
       opmath_t product = c10::detail::pi<opmath_t>() * opmath_t{a};
       if constexpr (c10::is_complex<opmath_t>::value) {
-        return static_cast<scalar_t>(std::sin(product) / product);
+        return static_cast<scalar_t>(c10::math::sin(product) / product);
       } else {
         return static_cast<scalar_t>(sycl::sin(product) / product);
       }
